src/components/ThrottleService.js

```diff
@@ -0,0 +1,55 @@
+(function() {
+  goog.provide('ga_throttle_service');
+
+  var module = angular.module('ga_throttle_service', []);
+
+  /**
+   *
+   * Heavily inspired from mrgamer's angular-throttle.js
+   * https://gist.github.com/mrgamer
+   *
+   */
+
+  //no deferring at the moment
+
+  module.provider('gaThrottle', function() {
+    this.$get = function($timeout) {
+      var Throttle = function() {
+        this.throttle = function(callback, delay, no_trailing, debounce_mode) {
```

Comment:
and can you add liek in gaDebounce a invokeApply parameter [1] pls.

[1] https://github.com/geoadmin/mf-geoadmin3/blob/master/src/components/DebounceService.js#L9